File: Py/cs50/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int shift(char c)
{
    if (c >= 65 && c <= 90)
    {
        c -= 65;
    }
    else if (c >= 97 && c <= 122)
    {
        c -= 97;
    }
    return c;
}

int main(int argC, string argV[])
{
    if (argC != 2)
    {
        printf("Wrong number of arguments");
        return 1;
    }
    for (int j = 0; j < strlen(argV[1]); j++)
    {
        if (!isalpha(argV[1][j]))
        {
            printf("Usage: ./vignere keyword");
            return 1;
        }
    }
    string k = argV[1];
    int kLength = strlen(k);
    string s = get_string("plaintext: ");
    int sLength = strlen(s);
    for (int i = 0, j = 0; i <= strlen(s); i++)
    {
        k[j] = shift(k[j % kLength]);
        if (s[i] >= 65 && s[i] <= 90)
        {
            s[i] = (s[i] + k[j] - 65) % 26 + 65;
            j++;
        }
        else if (s[i] >= 97 && s[i] <= 122)
        {
            s[i] = (s[i] + k[j] - 97) % 26 + 97;
            printf("%i\n", s[i]);
            j++;
        }
        else
        {
            s[i] = s[i];
        }
    }
    printf("ciphertext: %s\n", s);
    return 0;
}

File: Py/cs50/pset6/vigenere.py
import sys
from cs50 import get_string

# get key as commandline argument
k = sys.argv[1]
if len(sys.argv) != 2 or not k.isalpha():
    sys.exit("Usage: python vigenere.py key")

# Make key into adjusted list
listK = list(k)
for n, i in enumerate(listK):
    listK[n] = ord(i)
    if listK[n] >= 65 and listK[n] <= 90:
        listK[n] -= 65
    elif listK[n] >= 97 and listK[n] <= 122:
        listK[n] -= 97

# get plaintext
p = get_string("plaintext: ")

# for character in plaintext, get the # associated
listP = []
for c in p:
    listP.append((ord(c)))

# shift each # by key, wrapping if necessary
for n, i in enumerate(listP):
    if i >= 65 and i <= 90:
        listP[n] = (i + listK[n % len(listK)] - 65) % 26 + 65
    elif i >= 97 and i <= 122:
        listP[n] = (i + listK[n % len(listK)] - 97) % 26 + 97
    else:
        listP[n] = listP[n]

# make into list of chars, stringify, print
listC = []
for i in listP:
    listC.append(chr(i))
print(f"ciphertext: {''.join(listC)}")
